### DISCUSSION TITLE
Font glyphs not aligning with cursor

### DISCUSSION THREAD

**Discussion by caraar12345** [blinksh/blink#1643] [Support]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration

- Blink version: `16.0.2`
- iOS version: `16.1`

### Describe the bug

I'm using a custom [Input font](https://input.djr.com/) with wider glyphs than usual. Most renderers (VSCode, iTerm2 etc) are fine with this but Blink fails to align the cursor with the actual text. As a result, it's very difficult to edit commands etc as it's a guessing game as to where the cursor **actually** is.

https://aaroncarson.co.uk/blink/fonts.css

#### Expected behaviour

The text lines up with the cursor.

#### Screen recording

_Sorry the text is so small!_

https://user-images.githubusercontent.com/6129428/202702659-a789b3f4-fa1a-4118-8d91-34fb3202272f.mov

**Comment by thegcat** [comment]
I have a similar issue with the Fira fonts, installed as https://github.com/blinksh/fonts/pull/22

Is this a known problem? Is there a way to help debug this?

**Reply by carloscabanero** [reply]
There should be no issue with Fire. Can you send a screenshot? What OS are you on?

**Reply by thegcat** [reply]
I was the same as in the video above, I don't currently have this issue though, so I'd say this is solved for me.

In January I should have been on stable iPadOS and TestFlight Blink already I think, if it helps any.

**Comment by avysk** [comment]
I have tried to create a snippet to use "MSX Screen 0" font and I have the same issue. See https://gist.githubusercontent.com/avysk/ea238d6acab079768061d87bdddf9c22/raw/9eccc05d525ad13ba606dd8c1866fcda451ac9e3/msxfont.css

**Comment by carloscabanero** [comment]
Input comes in "proportional" (like Arial, etc...) and "monospaced" (the usual terminal ones) variants. In this context there are no 'wider glyphs than usual' unless the monospaced font is not really monospaced. Please note that this has nothing to do with double width glyphs and all that, which we support. I think you may be used the "proportional" font variant, and Blink will not support that.

A few years ago when we reworked rendering for Hterm we had to decide how flexible we wanted to be with non-monospaced fonts. As those cases are rare, we decided to go for speed and rely on monospaced fonts being... monospaced and doing the right thing on their own for 95% of the cases. I still think that being a terminal this is what makes sense (as then there are vertical lines in editors, etc...).

Let us know if you still have the same issue with the Monospaced font, and will take a look.

**Comment by avysk** [comment]
I use the latest Blink from AppStore on the latest iPadOS 16. The font in question is monospaced. I have given above the URL you can use for its installation.

**Comment by yury** [comment]
Hello @caraar12345. I checked your font. Couple issues there. Font families are different and font should be base64 encoded.
Please try [this one](https://gist.githubusercontent.com/yury/fc21967b8e0ebda0679623d1f4775241/raw/4f6e8ac4ca57cee8b3e94756e5070e83f7ad5c7f/input.css).

**Comment by yury** [comment]
Hello @avysk 

Please try [this one](https://gist.githubusercontent.com/yury/4ba142bc6b148ea0770c6bd8a8424b74/raw/1c60d630b5759ce193d063cf82d013ef493f4002/msx-screen.css)

Same thing. It should be base64 encoded. Also I added bold one.

**Reply by avysk** [reply]
Thank you very much, it works now.

However, I'd like to notice that the whole story is quite annoying: Windows, Linux and FreeBSD are happy with the font; it works fine in different terminal emulators and in gvim -- but Blink wants something different (even if it wants "the right thing").
